In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

# Use the kagglehub client library to attach Kaggle resources like competitions, datasets, and models to your session
# Learn more about kagglehub: https://github.com/Kaggle/kagglehub/blob/main/README.md

import kagglehub
# kagglehub.dataset_download('<owner>/<dataset-slug>')

/kaggle/input/datasets/yelp-dataset/yelp-dataset/Dataset_User_Agreement.pdf
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_review.json
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_checkin.json
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_business.json
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_tip.json
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_user.json


In [2]:
import os

for root, dirs, files in os.walk("/kaggle/input"):
    level = root.replace("/kaggle/input", "").count(os.sep)
    indent = "  " * level
    print(f"{indent}{os.path.basename(root)}/")
    for file in files[:20]:
        print(f"{indent}  {file}")

input/
  datasets/
    yelp-dataset/
      yelp-dataset/
        Dataset_User_Agreement.pdf
        yelp_academic_dataset_review.json
        yelp_academic_dataset_checkin.json
        yelp_academic_dataset_business.json
        yelp_academic_dataset_tip.json
        yelp_academic_dataset_user.json


In [3]:
DATA_PATH = "/kaggle/input/datasets/yelp-dataset/yelp-dataset"

BUSINESS_PATH = f"{DATA_PATH}/yelp_academic_dataset_business.json"
REVIEW_PATH = f"{DATA_PATH}/yelp_academic_dataset_review.json"
CHECKIN_PATH = f"{DATA_PATH}/yelp_academic_dataset_checkin.json"

print(BUSINESS_PATH)
print(REVIEW_PATH)
print(CHECKIN_PATH)

/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_business.json
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_review.json
/kaggle/input/datasets/yelp-dataset/yelp-dataset/yelp_academic_dataset_checkin.json


In [4]:
import os

print("Business:", os.path.exists(BUSINESS_PATH))
print("Review:", os.path.exists(REVIEW_PATH))
print("Checkin:", os.path.exists(CHECKIN_PATH))

Business: True
Review: True
Checkin: True


In [5]:
#load business.json
import pandas as pd
import numpy as np

business = pd.read_json(
    BUSINESS_PATH,
    lines=True
)

print("Shape:", business.shape)
print("\nColumns:")
print(business.columns.tolist())

Shape: (150346, 14)

Columns:
['business_id', 'name', 'address', 'city', 'state', 'postal_code', 'latitude', 'longitude', 'stars', 'review_count', 'is_open', 'attributes', 'categories', 'hours']


In [6]:
business.head()

,business_id,name,address,city,state,postal_code,latitude,longitude,stars,review_count,is_open,attributes,categories,hours
0,Pns2l4eNsfO8kk83dixA6A,"Abby Rappoport, LAC, CMQ","1616 Chapala St, Ste 2",Santa Barbara,CA,93101,34.426679,-119.711197,5.0,7,0,{'ByAppointmentOnly': 'True'},"Doctors, Traditional Chinese Medicine, Naturop...",None
1,mpf3x-BjTdTEA3yCZrAYPw,The UPS Store,87 Grasso Plaza Shopping Center,Affton,MO,63123,38.551126,-90.335695,3.0,15,1,{'BusinessAcceptsCreditCards': 'True'},"Shipping Centers, Local Services, Notaries, Ma...","{'Monday': '0:0-0:0', 'Tuesday': '8:0-18:30', ..."
2,tUFrWirKiKi_TAnsVWINQQ,Target,5255 E Broadway Blvd,Tucson,AZ,85711,32.223236,-110.880452,3.5,22,0,"{'BikeParking': 'True', 'BusinessAcceptsCredit...","Department Stores, Shopping, Fashion, Home & G...","{'Monday': '8:0-22:0', 'Tuesday': '8:0-22:0', ..."
3,MTSW4McQd7CbVtyjqoe9mw,St Honore Pastries,935 Race St,Philadelphia,PA,19107,39.955505,-75.155564,4.0,80,1,"{'RestaurantsDelivery': 'False', 'OutdoorSeati...","Restaurants, Food, Bubble Tea, Coffee & Tea, B...","{'Monday': '7:0-20:0', 'Tuesday': '7:0-20:0', ..."
4,mWMc6_wTdE0EUBKIGXDVfA,Perkiomen Valley Brewery,101 Walnut St,Green Lane,PA,18054,40.338183,-75.471659,4.5,13,1,"{'BusinessAcceptsCreditCards': 'True', 'Wheelc...","Brewpubs, Breweries, Food","{'Wednesday': '14:0-22:0', 'Thursday': '16:0-2..."


In [7]:
business.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150346 entries, 0 to 150345
Data columns (total 14 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   business_id   150346 non-null  object 
 1   name          150346 non-null  object 
 2   address       150346 non-null  object 
 3   city          150346 non-null  object 
 4   state         150346 non-null  object 
 5   postal_code   150346 non-null  object 
 6   latitude      150346 non-null  float64
 7   longitude     150346 non-null  float64
 8   stars         150346 non-null  float64
 9   review_count  150346 non-null  int64  
 10  is_open       150346 non-null  int64  
 11  attributes    136602 non-null  object 
 12  categories    150243 non-null  object 
 13  hours         127123 non-null  object 
dtypes: float64(3), int64(2), object(9)
memory usage: 16.1+ MB


In [8]:
#check the categories column
print(business["categories"].head(10).to_string())

0    Doctors, Traditional Chinese Medicine, Naturop...
1    Shipping Centers, Local Services, Notaries, Ma...
2    Department Stores, Shopping, Fashion, Home & G...
3    Restaurants, Food, Bubble Tea, Coffee & Tea, B...
4                            Brewpubs, Breweries, Food
5    Burgers, Fast Food, Sandwiches, Food, Ice Crea...
6    Sporting Goods, Fashion, Shoe Stores, Shopping...
7                  Synagogues, Religious Organizations
8    Pubs, Restaurants, Italian, Bars, American (Tr...
9    Ice Cream & Frozen Yogurt, Fast Food, Burgers,...


In [9]:
print("Missing categories:", business["categories"].isna().sum())

Missing categories: 103


In [10]:
#Find the cities with the most restaurants
restaurant_business = business[
    business["categories"].fillna("").str.contains(
        "Restaurants",
        case=False,
        regex=False
    )
].copy()

print("Restaurant businesses:", len(restaurant_business))

Restaurant businesses: 52268


In [11]:
restaurant_city_counts = (
    restaurant_business["city"]
    .value_counts()
    .reset_index()
)

restaurant_city_counts.columns = [
    "city",
    "restaurant_count"
]

print(
    restaurant_city_counts
    .head(30)
    .to_string(index=False)
)

            city  restaurant_count
    Philadelphia              5852
           Tampa              2960
    Indianapolis              2862
       Nashville              2502
          Tucson              2466
     New Orleans              2259
        Edmonton              2166
     Saint Louis              1790
            Reno              1286
           Boise               847
   Santa Barbara               767
      Clearwater               678
      Wilmington               619
       St. Louis               543
        Metairie               522
Saint Petersburg               491
        Franklin               444
  St. Petersburg               404
          Sparks               334
         Brandon               326
        Meridian               312
           Largo               310
     Cherry Hill               297
          Carmel               291
    West Chester               280
          Kenner               228
 New Port Richey               226
          Goleta    

In [12]:
#check filtered business count
SELECTED_CITIES = ["Philadelphia", "Tampa"]

yelp_business_sample = business[
    business["categories"].fillna("").str.contains(
        "Restaurants",
        case=False,
        regex=False
    )
    & business["city"].isin(SELECTED_CITIES)
].copy()

print("Filtered business count:", len(yelp_business_sample))
print("\nBusinesses by city:")
print(yelp_business_sample["city"].value_counts())

Filtered business count: 8812

Businesses by city:
city
Philadelphia    5852
Tampa           2960
Name: count, dtype: int64


In [13]:
#save the business sample
business_output = "/kaggle/working/yelp_business_sample.csv"

yelp_business_sample.to_csv(
    business_output,
    index=False
)

print("Saved:", business_output)

Saved: /kaggle/working/yelp_business_sample.csv


In [14]:
import os

print("File exists:", os.path.exists(business_output))
print(
    "File size:",
    round(os.path.getsize(business_output) / (1024 ** 2), 2),
    "MB"
)

File exists: True
File size: 8.59 MB


In [15]:
#create the business_id set
business_ids = set(
    yelp_business_sample["business_id"]
)

print("Business rows:", len(yelp_business_sample))
print("Unique business IDs:", len(business_ids))

Business rows: 8812
Unique business IDs: 8812


In [16]:
#verify there are no duplicates
print(
    "Duplicate business IDs:",
    yelp_business_sample["business_id"].duplicated().sum()
)

Duplicate business IDs: 0


In [18]:
CHUNK_SIZE = 100_000

review_chunks = []

total_rows_processed = 0
total_reviews_matched = 0

In [19]:
for i, chunk in enumerate(
    pd.read_json(
        REVIEW_PATH,
        lines=True,
        chunksize=CHUNK_SIZE
    )
):
    
    total_rows_processed += len(chunk)

    matched = chunk[
        chunk["business_id"].isin(business_ids)
    ]

    if not matched.empty:
        review_chunks.append(matched)
        total_reviews_matched += len(matched)

    if (i + 1) % 10 == 0:
        print(
            f"Chunks processed: {i + 1:,} | "
            f"Rows processed: {total_rows_processed:,} | "
            f"Matched reviews: {total_reviews_matched:,}"
        )

Chunks processed: 10 | Rows processed: 1,000,000 | Matched reviews: 149,704
Chunks processed: 20 | Rows processed: 2,000,000 | Matched reviews: 294,211
Chunks processed: 30 | Rows processed: 3,000,000 | Matched reviews: 439,097
Chunks processed: 40 | Rows processed: 4,000,000 | Matched reviews: 570,962
Chunks processed: 50 | Rows processed: 5,000,000 | Matched reviews: 707,673
Chunks processed: 60 | Rows processed: 6,000,000 | Matched reviews: 851,662
Chunks processed: 70 | Rows processed: 6,990,280 | Matched reviews: 990,521


In [20]:
yelp_review_sample = pd.concat(
    review_chunks,
    ignore_index=True
)

print("Review sample shape:", yelp_review_sample.shape)
print("Review count:", len(yelp_review_sample))

Review sample shape: (990521, 9)
Review count: 990521


In [21]:
print("Businesses represented in reviews:",
      yelp_review_sample["business_id"].nunique())

review_business_ids = set(
    yelp_review_sample["business_id"]
)

orphan_review_ids = review_business_ids - business_ids

print("Orphan review business IDs:",
      len(orphan_review_ids))

Businesses represented in reviews: 8812
Orphan review business IDs: 0


In [22]:
review_output = "/kaggle/working/yelp_review_sample.parquet"

yelp_review_sample.to_parquet(
    review_output,
    index=False,
    compression="snappy"
)

print("Saved:", review_output)

Saved: /kaggle/working/yelp_review_sample.parquet


In [23]:
import os

print("File exists:", os.path.exists(review_output))
print(
    "File size:",
    round(os.path.getsize(review_output) / (1024 ** 2), 2),
    "MB"
)

File exists: True
File size: 400.38 MB


In [24]:
#load checkin.json
checkin = pd.read_json(
    CHECKIN_PATH,
    lines=True
)

print("Checkin shape:", checkin.shape)
print("Columns:", checkin.columns.tolist())

Checkin shape: (131930, 2)
Columns: ['business_id', 'date']


In [25]:
checkin.head()

,business_id,date
0,---kPU91CF4Lq2-WlRu9Lw,"2020-03-13 21:10:56, 2020-06-02 22:18:06, 2020..."
1,--0iUa4sNDFiZFrAdIWhZQ,"2010-09-13 21:43:09, 2011-05-04 23:08:15, 2011..."
2,--30_8IhuyMHbSOcNWd6DQ,"2013-06-14 23:29:17, 2014-08-13 23:20:22"
3,--7PUidqRWpRSpXebiyxTg,"2011-02-15 17:12:00, 2011-07-28 02:46:10, 2012..."
4,--7jw19RH9JKXgFohspgQw,"2014-04-21 20:42:11, 2014-04-28 21:04:46, 2014..."


In [26]:
yelp_checkin_sample = checkin[
    checkin["business_id"].isin(business_ids)
].copy()

print(
    "Filtered checkin rows:",
    len(yelp_checkin_sample)
)

Filtered checkin rows: 8583


In [27]:
#check for orphan ids
checkin_business_ids = set(
    yelp_checkin_sample["business_id"]
)

orphan_checkin_ids = (
    checkin_business_ids - business_ids
)

print(
    "Orphan checkin business IDs:",
    len(orphan_checkin_ids)
)

Orphan checkin business IDs: 0


In [28]:
#check how many business have checkins
print(
    "Businesses represented in checkins:",
    yelp_checkin_sample["business_id"].nunique()
)

Businesses represented in checkins: 8583


In [29]:
#inspect a few rows
yelp_checkin_sample.head()

,business_id,date
48,-0M0b-XhtFagyLmsBtOe8w,"2012-02-09 22:11:59, 2012-02-17 23:03:05, 2012..."
49,-0PN_KFPtbnLQZEeb23XiA,"2011-02-23 05:39:14, 2011-03-01 21:58:33, 2011..."
50,-0TffRSXXIlBYVbb5AwfTg,"2013-05-31 23:03:18, 2013-05-31 23:05:01, 2013..."
55,-0eUa8TsXFFy0FCxHYmrjg,"2015-09-25 17:19:51, 2015-09-28 18:00:56, 2015..."
62,-0i2KNr7WrCsDF5m0IViJg,"2010-03-25 17:00:35, 2010-08-11 17:50:29, 2010..."


In [31]:
yelp_checkin_sample.sample(
    min(5, len(yelp_checkin_sample)),
    random_state=42
)

,business_id,date
77066,_PJRuNpXHmPCv9eGXA3sxg,"2010-11-19 22:16:54, 2011-01-24 19:55:09, 2011..."
103185,mB0UiTwtgSZdMERifbfjFA,"2013-02-10 03:04:40, 2013-02-12 01:37:08, 2013..."
103755,mS_B6-fEgGSpsPTYELgSYQ,"2010-09-18 20:59:03, 2010-10-24 16:55:33, 2010..."
3494,0dfspevYKsugYiU1UVl94Q,"2018-02-24 17:25:22, 2018-03-25 21:42:39, 2018..."
85812,dikP2bqi-zIUISB_TKPRtA,"2018-09-19 22:22:27, 2018-09-22 18:42:57, 2018..."


In [32]:
#save the checkin sample
checkin_output = "/kaggle/working/yelp_checkin_sample.csv"

yelp_checkin_sample.to_csv(
    checkin_output,
    index=False
)

print("Saved:", checkin_output)

Saved: /kaggle/working/yelp_checkin_sample.csv


In [33]:
#verify
import os

print(
    "File exists:",
    os.path.exists(checkin_output)
)

print(
    "File size:",
    round(
        os.path.getsize(checkin_output) / (1024 ** 2),
        2
    ),
    "MB"
)

File exists: True
File size: 36.18 MB


In [35]:
#final sanity checks:

#Business Sample
print("Business count:", len(yelp_business_sample))
print(
    "Unique business IDs:",
    yelp_business_sample["business_id"].nunique()
)

#Review Sample
print("Review count:", len(yelp_review_sample))
print(
    "Businesses represented in reviews:",
    yelp_review_sample["business_id"].nunique()
)

#Checkin Sample
print("Checkin rows:", len(yelp_checkin_sample))
print(
    "Businesses represented in checkins:",
    yelp_checkin_sample["business_id"].nunique()
)

#Orphan checks
print(
    "Orphan review IDs:",
    len(orphan_review_ids)
)

print(
    "Orphan checkin IDs:",
    len(orphan_checkin_ids)
)

Business count: 8812
Unique business IDs: 8812
Review count: 990521
Businesses represented in reviews: 8812
Checkin rows: 8583
Businesses represented in checkins: 8583
Orphan review IDs: 0
Orphan checkin IDs: 0


In [36]:
#spot check the actual data:

#Business sample:
yelp_business_sample[
    [
        "business_id",
        "name",
        "city",
        "state",
        "stars",
        "review_count",
        "categories"
    ]
].sample(
    5,
    random_state=42
)

,business_id,name,city,state,stars,review_count,categories
67432,JwrIiF4oihcg9rjyHjxY7w,Capri Pizza-N-More,Tampa,FL,4.5,63,"Pasta Shops, Pizza, Restaurants, Food, Special..."
110256,QS884ZcHmkTaWAmjrXyFlg,Miya Lunch Box and Catering,Philadelphia,PA,4.0,5,"Japanese, Restaurants"
108802,I30Nq1Y8MKcFmQNeYST36Q,Chick-fil-A,Tampa,FL,3.0,96,"Restaurants, Breakfast & Brunch, Event Plannin..."
3655,x2J-YIFeGZ-nsezzooVA9g,Twenty 21,Philadelphia,PA,3.0,8,Restaurants
149598,YaWt65tSRsyK6bppZUY-dg,Bernie's Deli,Tampa,FL,2.5,9,"Restaurants, American (Traditional), American ..."


In [37]:
#spot check the actual data:

#Review sample:
yelp_review_sample[
    [
        "review_id",
        "user_id",
        "business_id",
        "stars",
        "text",
        "date"
    ]
].sample(
    5,
    random_state=42
)

,review_id,user_id,business_id,stars,text,date
184759,ND25atwxJ_yZb3J2HSrdfw,wgBrfXIy51fQ0zjmi7UudA,DTwt8_pCuGyQbiIfXNdhDw,1,Just went to the Barnes and Noble an hour ago ...,2020-02-27 08:20:17
650703,im5Sy0wAIoowkiLUS4pqzg,7igTVTgqI3WiqZp8hrgaMA,un0xxX785wfQfprryh0TjQ,4,Would easily rate five stars if the South Phil...,2011-12-31 17:00:49
35420,zc1KKE4TBdYJArqd4s4HjA,QDvOUvgNocu8qcW6OrmnfQ,kkcQYuF3w5iHnHMf0EnRhQ,5,AMAZING!!! Milkshake flavors of the day...Sant...,2016-12-02 18:45:41
35473,u9RRp0O_c1VyDOlh4TcGbg,JSbxxfelWwGRY3duM8htGA,i_5PaxI8DK9EErsHBdwP1A,4,Personal-size murgh makhani pizza for $5. Get ...,2010-12-03 17:25:57
423649,IDiD3jPutQJlthxmd3wVvA,mJtYMmLcpofP7qBK1DfZQA,HxSAlo0GRb4JTmDIxNw68Q,2,"The pizza was pretty good, the service was sho...",2017-08-11 23:46:27


In [38]:
#spot check the actual data:

#Checkin sample:
yelp_checkin_sample.sample(
    5,
    random_state=42
)

,business_id,date
77066,_PJRuNpXHmPCv9eGXA3sxg,"2010-11-19 22:16:54, 2011-01-24 19:55:09, 2011..."
103185,mB0UiTwtgSZdMERifbfjFA,"2013-02-10 03:04:40, 2013-02-12 01:37:08, 2013..."
103755,mS_B6-fEgGSpsPTYELgSYQ,"2010-09-18 20:59:03, 2010-10-24 16:55:33, 2010..."
3494,0dfspevYKsugYiU1UVl94Q,"2018-02-24 17:25:22, 2018-03-25 21:42:39, 2018..."
85812,dikP2bqi-zIUISB_TKPRtA,"2018-09-19 22:22:27, 2018-09-22 18:42:57, 2018..."


In [39]:
#generate the final project summary
final_summary = pd.DataFrame({
    "metric": [
        "Selected cities",
        "Restaurant businesses",
        "Philadelphia businesses",
        "Tampa businesses",
        "Reviews",
        "Businesses represented in reviews",
        "Checkin rows",
        "Businesses represented in checkins",
        "Orphan review business IDs",
        "Orphan checkin business IDs"
    ],
    "value": [
        ", ".join(SELECTED_CITIES),
        len(yelp_business_sample),
        (yelp_business_sample["city"] == "Philadelphia").sum(),
        (yelp_business_sample["city"] == "Tampa").sum(),
        len(yelp_review_sample),
        yelp_review_sample["business_id"].nunique(),
        len(yelp_checkin_sample),
        yelp_checkin_sample["business_id"].nunique(),
        len(orphan_review_ids),
        len(orphan_checkin_ids)
    ]
})

final_summary

,metric,value
0,Selected cities,"Philadelphia, Tampa"
1,Restaurant businesses,8812
2,Philadelphia businesses,5852
3,Tampa businesses,2960
4,Reviews,990521
5,Businesses represented in reviews,8812
6,Checkin rows,8583
7,Businesses represented in checkins,8583
8,Orphan review business IDs,0
9,Orphan checkin business IDs,0


## Yelp Sampling Summary

**Source:** Yelp Open Dataset

**Business population:**
- Category filter: categories contains "Restaurants"
- Cities: Philadelphia and Tampa
- Businesses: 8,812

**Review population:**
- Reviews linked to the 8,812 sampled businesses
- Reviews retained: 990,521
- All 8,812 sampled businesses are represented

**Check-in population:**
- Check-in records linked to the sampled businesses
- Check-in rows/businesses represented: 8,583
- 229 sampled businesses have no check-in record

**Data quality:**
- Orphan review business IDs: 0
- Orphan check-in business IDs: 0

## Review Sample Download Preparation

The filtered review sample contains 990,521 reviews and is stored as a single in-memory DataFrame. Because the resulting Parquet file is too large for reliable direct download from the Kaggle interface, it is partitioned into smaller Parquet files for transfer to the local environment.

The partitions are temporary transfer files and will be recombined locally into a single `yelp_review_sample.parquet` dataset.

In [40]:
import os
import glob

output_dir = "/kaggle/working/yelp_review_parts"
os.makedirs(output_dir, exist_ok=True)

# Remove previously generated parts
for f in glob.glob(f"{output_dir}/*.parquet"):
    os.remove(f)

CHUNK_ROWS = 100_000

for i, start in enumerate(
    range(0, len(yelp_review_sample), CHUNK_ROWS),
    start=1
):
    end = min(start + CHUNK_ROWS, len(yelp_review_sample))

    part = yelp_review_sample.iloc[start:end]

    output_path = (
        f"{output_dir}/"
        f"yelp_review_sample_part_{i:02d}.parquet"
    )

    part.to_parquet(
        output_path,
        index=False,
        compression="snappy"
    )

    size_mb = os.path.getsize(output_path) / (1024 ** 2)

    print(
        f"Part {i:02d}: "
        f"{len(part):,} rows | "
        f"{size_mb:.2f} MB"
    )

Part 01: 100,000 rows | 40.82 MB
Part 02: 100,000 rows | 40.47 MB
Part 03: 100,000 rows | 40.00 MB
Part 04: 100,000 rows | 40.82 MB
Part 05: 100,000 rows | 40.68 MB
Part 06: 100,000 rows | 40.56 MB
Part 07: 100,000 rows | 40.69 MB
Part 08: 100,000 rows | 39.61 MB
Part 09: 100,000 rows | 40.65 MB
Part 10: 90,521 rows | 36.02 MB


In [42]:
parts = sorted(
    glob.glob(
        "/kaggle/working/yelp_review_parts/*.parquet"
    )
)

print("\nTotal parts:", len(parts))

for p in parts:
    print(
        os.path.basename(p),
        "→",
        round(os.path.getsize(p) / (1024 ** 2), 2),
        "MB"
    )


Total parts: 10
yelp_review_sample_part_01.parquet → 40.82 MB
yelp_review_sample_part_02.parquet → 40.47 MB
yelp_review_sample_part_03.parquet → 40.0 MB
yelp_review_sample_part_04.parquet → 40.82 MB
yelp_review_sample_part_05.parquet → 40.68 MB
yelp_review_sample_part_06.parquet → 40.56 MB
yelp_review_sample_part_07.parquet → 40.69 MB
yelp_review_sample_part_08.parquet → 39.61 MB
yelp_review_sample_part_09.parquet → 40.65 MB
yelp_review_sample_part_10.parquet → 36.02 MB
